pkg/frontend/admin_openshiftcluster_resources_list.go

```diff
@@ -88,6 +91,29 @@ func (f *frontend) _listAdminOpenShiftClusterResources(ctx context.Context, r *h
 			Resource: gr,
 		})
 	}
+	// enrich will not return error
+	f.enrich(ctx, f.baseLog, &armResources, resource, clusterResourceGroup, fpAuthorizer)
 
 	return json.Marshal(armResources)
 }
+
+// enrich will add additional data to the arm output
+func (f *frontend) enrich(ctx context.Context, log *logrus.Entry, armResources *[]arm.Resource, resource azure.Resource, clusterResourceGroup string, fpAuthorizer autorest.Authorizer) {
```

Comment:
why not just roll this into the function above such that if it's a VM you call vmCli.Get, otherwise call resourcesClient.GetByID?
less scary type casting that way.